simulators/hwp/hwp_emulator.py

```diff
@@ -0,0 +1,192 @@
+"""
+HWP Emulation module
+"""
+import logging
+import threading
+import time
+from copy import deepcopy
+from dataclasses import dataclass
+
+from socs.testing import device_emulator
+
+
+def hex_str_to_dec(hex_value, decimal=3):
+    """Converts a hex string to a decimal float"""
+    return float(int(hex_value, 16)) / 10**decimal
+
+
+def float_to_hex_str(value, decimal=3):
+    """Converts a decimal float to a hex string"""
+    return "0000" + str(hex(int(value * 10**decimal)))[2:]
```

Comment:
I realized this was a (incomplete) copy of [this function from the PID driver](https://github.com/simonsobs/socs/blame/fba087ec34401fde9251e62d072d9a1f081a8579/socs/agents/hwp_pid/drivers/pid_controller.py#L63-L70), so I think this can be replaced entirely. The driver function does not use zfill, but does take just 4 hex digits, which enforces a string length of 8. I'm not sure if this is an ideal implementation, but I think its probably a good idea to just use the driver fn.